src/index.ts

```diff
@@ -1,33 +1,36 @@
-import { Users } from './services/users';
-import { Logger } from './services/logger';
+import type { User } from "./types";
 
-import type { User, ApiConfig } from './types';
+import { createIoCContainer } from "./ioc";
 
-const renderUsers = async (config: ApiConfig) => {
-  const usersService = new Users(config);
+import IoCContainer from "ioc-lite";
+
+const renderUsers = async (ioc: IoCContainer) => {
+  const usersService = ioc.resolve("user");
   const users = await usersService.getUsers();
 
-  const listNode = document.getElementById('users-list');
+  const listNode = document.getElementById("users-list");
 
   users.forEach((user: User) => {
-    const listItemNode = document.createElement('li');
+    const listItemNode = document.createElement("li");
 
     listItemNode.innerHTML = user.name;
     listNode.appendChild(listItemNode);
   });
 };
 
-const app = () => {
-  const config = (window as any).__CONFIG__;
+const app = (ioc: IoCContainer) => {
   delete (window as any).__CONFIG__;
 
-  renderUsers(config.api);
+  renderUsers(ioc);
 };
 
 window.onload = (event: Event) => {
-  const logger = new Logger();
+  const config = (window as any).__CONFIG__;
+  const ioc = createIoCContainer(config.api);
```

Comment:
It's better to move the `ioc container` to the root of the file and even add export to allow import and use it in different places. In this case, you can register and resolve your resources whenever you want.